### DISCUSSION TITLE
Objects like simple values in expressions

### DISCUSSION THREAD

**Discussion by Big-Gremlin** [josdejong/mathjs#3065] [Q&A]
Is it possible to use objects like simple values?

Example:
~~~~
export class CalcObj {
    value: number;
    filtered: boolean;

    constructor(value: number) {
        this.value = value;
    }
}

const scope = {
    objA: new CalcObj(3),
    objB: new CalcObj(2)
}

const result = math.eval('objA + objB', scope); 
# result = 5
~~~~

Can a special function be used, such as toString () or something similar?

~~~~
export class CalcObj {
    public value: number;
    public filtered: boolean;

    constructor(value: number) {
        this.value = value;
    }

   public toString(): string {
       return this.value
   }
}
~~~~

**Comment by josdejong** [comment]
It is possible to add support for a new data type and do calculations with it in the expression parser. You can find a small demo here:

https://github.com/josdejong/mathjs/blob/develop/examples/advanced/custom_datatype.js

is that what you mean?

**Comment by gwhitney** [comment]
This is a Question, rather than an issue. Moving.

**Comment by gwhitney** [comment]
There is also a further question here. Right now for arrays we have `math.add([1, 2], [3, 4])` returns `[4,6]`, but we don't have anything similar for `math.add({x: 1, y: 2}, {x: 3, y: 4})` -- it throws an unexpected type of argument. We could contemplate that at least for some simple operations, if you perform them on two objects, the operations "distribute" down to the like-named properties. If we implemented something like that, this function call would produce `{x: 4, y: 6}`, which could definitely at times be handy.
